src/global/global_init.cc

```diff
@@ -166,106 +166,107 @@ global_init(std::vector < const char * > *alt_def_args,
   if (g_conf->log_flush_on_exit)
     g_ceph_context->_log->set_flush_on_exit();
 
+  ostringstream priv_ss;
+
   // consider --setuser root a no-op, even if we're not root
   if (getuid() != 0) {
     if (g_conf->setuser.length()) {
       cerr << "ignoring --setuser " << g_conf->setuser << " since I am not root"
 	   << std::endl;
-      g_conf->set_val("setuser", "", false);
     }
     if (g_conf->setgroup.length()) {
       cerr << "ignoring --setgroup " << g_conf->setgroup
 	   << " since I am not root" << std::endl;
-      g_conf->set_val("setgroup", "", false);
     }
   }
-
-  // drop privileges?
-  ostringstream priv_ss;
-  if (g_conf->setgroup.length() ||
-      g_conf->setuser.length()) {
-    uid_t uid = 0;  // zero means no change; we can only drop privs here.
-    gid_t gid = 0;
-    std::string uid_string;
-    std::string gid_string;
-    if (g_conf->setuser.length()) {
-      uid = atoi(g_conf->setuser.c_str());
-      if (!uid) {
-	char buf[4096];
-	struct passwd pa;
-	struct passwd *p = 0;
-	getpwnam_r(g_conf->setuser.c_str(), &pa, buf, sizeof(buf), &p);
-	if (!p) {
-	  cerr << "unable to look up user '" << g_conf->setuser << "'"
-	       << std::endl;
-	  exit(1);
-	}
-	uid = p->pw_uid;
-	gid = p->pw_gid;
-	uid_string = g_conf->setuser;
+  else {
+    // drop privileges?
+
+    if (g_conf->setgroup.length() ||
+        g_conf->setuser.length()) {
```

Comment:
i think if you switch this to else if () { .. you won't have to reindent this whole block?